src/VisualStudio/Core/Def/KeybindingReset/KeybindingResetDetector.cs

```diff
@@ -163,10 +163,13 @@ private void StartUpdateStateMachine()
 
         // make sure all state machine change work is serialized so that cancellation
         // doesn't mess the state up.   
-        _lastTask = _lastTask.SafeContinueWithFromAsync(_ =>
+        _lastTask = ContinueAsync(_lastTask);
+
+        async Task ContinueAsync(Task task)
         {
-            return UpdateStateMachineWorkerAsync(cancellationToken);
-        }, cancellationToken, TaskScheduler.Default);
+            await task.NoThrowAwaitableInternal(captureContext: false);
```

Comment:
the prior logic always wanted to run UpdateStateMachineWorkerAsync no matter what happened with the prior task.  That's why we use NoThrowAwaitableInternal here.